sqlglot/dialects/snowflake.py

```diff
@@ -863,18 +863,22 @@ def _parse_file_location(self) -> t.Optional[exp.Expression]:
                 else self._parse_table_parts()
             )
 
-        def _parse_location_path(self) -> exp.Var:
-            parts = [self._advance_any(ignore_reserved=True)]
+        def _parse_location_path(self) -> exp.Literal:
+            start = self._curr
+            self._advance_any(ignore_reserved=True)
 
             # We avoid consuming a comma token because external tables like @foo and @bar
             # can be joined in a query with a comma separator, as well as closing paren
             # in case of subqueries
             while self._is_connected() and not self._match_set(
                 (TokenType.COMMA, TokenType.L_PAREN, TokenType.R_PAREN), advance=False
             ):
-                parts.append(self._advance_any(ignore_reserved=True))
+                self._advance_any(ignore_reserved=True)
 
-            return exp.var("".join(part.text for part in parts if part))
+            result = self._find_sql(start, self._prev)
+            if len(result) >= 2 and result[0] == result[-1] == "'":
```

Comment:
These the two lines (879-880) were required (see also the other comment on the PR), as the parsed `result` string can be either an unquoted (`@mystage/foo.txt`) or a quoted string (`'@mystage/foo.txt'`).

Happy for any guidance here if you have a better approach for this in mind!